sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpBuilderPatternCondition.cs

```diff
@@ -0,0 +1,96 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SonarAnalyzer.Helpers
+{
+    public class CSharpBuilderPatternCondition : BuilderPatternCondition<InvocationExpressionSyntax>
+    {
+
+        public CSharpBuilderPatternCondition(bool constructorIsSafe, params BuilderPatternDescriptor<InvocationExpressionSyntax>[] descriptors) : base(constructorIsSafe, descriptors) { }
+
+        protected override SyntaxNode RemoveParentheses(SyntaxNode node) =>
+            node.RemoveParentheses();
+
+        protected override SyntaxNode TopMostContainingMethod(SyntaxNode node) =>
+            node.GetTopMostContainingMethod();
```

Comment:
I find `GetTopMostContainingMethod` much better than `TopMostContainingMethod`.

Reply:
Only in this case, because the original extension does already contains redundant and unneeded Get prefix. The `Get` prefix is redundant to the `= ... ()` syntax 🙂